x-pack/plugins/security_solution/server/lib/telemetry/tasks/prebuilt_rule_alerts.ts

```diff
@@ -47,53 +52,56 @@ export function createTelemetryPrebuiltRuleAlertsTaskConfig(maxTelemetryBatch: n
-          await sender.sendOnDemand(TASK_METRICS_CHANNEL, [
-            createTaskMetric(taskName, true, startTime),
-          ]);
+        if (index === undefined) {
+          tlog(logger, `alerts index is not ready yet, skipping telemetry task`);
           return 0;
         }
 
-        const processedAlerts = telemetryEvents.map(
-          (event: TelemetryEvent): TelemetryEvent =>
-            copyAllowlistedFields(filterList.prebuiltRulesAlerts, event)
-        );
+        let fetchMore = true;
+        let searchAfterValue: SortResults | undefined;
+        let pitId = await receiver.openPointInTime(index);
+
+        while (fetchMore) {
+          const { moreToFetch, newPitId, searchAfter, alerts } =
+            await receiver.fetchPrebuiltRuleAlertsBatch(pitId, searchAfterValue);
 
-        const sanitizedAlerts = processedAlerts.map(
-          (event: TelemetryEvent): TelemetryEvent =>
-            processK8sUsernames(clusterInfo?.cluster_uuid, event)
-        );
+          fetchMore = moreToFetch;
+          searchAfterValue = searchAfter;
+          pitId = newPitId;
 
-        const enrichedAlerts = sanitizedAlerts.map(
-          (event: TelemetryEvent): TelemetryEvent => ({
-            ...event,
-            licence_id: licenseInfo?.uid,
-            cluster_uuid: clusterInfo?.cluster_uuid,
-            cluster_name: clusterInfo?.cluster_name,
-            package_version: packageInfo?.version,
-          })
-        );
+          const processedAlerts = alerts.map(
+            (event: TelemetryEvent): TelemetryEvent =>
+              copyAllowlistedFields(filterList.prebuiltRulesAlerts, event)
+          );
 
-        tlog(logger, `sending ${enrichedAlerts.length} elastic prebuilt alerts`);
-        const batches = batchTelemetryRecords(enrichedAlerts, maxTelemetryBatch);
-        for (const batch of batches) {
-          await sender.sendOnDemand(TELEMETRY_CHANNEL_DETECTION_ALERTS, batch);
+          const sanitizedAlerts = processedAlerts.map(
+            (event: TelemetryEvent): TelemetryEvent =>
+              processK8sUsernames(clusterInfo?.cluster_uuid, event)
+          );
+
+          const enrichedAlerts = sanitizedAlerts.map(
+            (event: TelemetryEvent): TelemetryEvent => ({
+              ...event,
+              licence_id: licenseInfo?.uid,
+              cluster_uuid: clusterInfo?.cluster_uuid,
+              cluster_name: clusterInfo?.cluster_name,
+              package_version: packageInfo?.version,
+              task_version: taskVersion,
+            })
+          );
+
+          tlog(logger, `sending ${enrichedAlerts.length} elastic prebuilt alerts`);
+          const batches = batchTelemetryRecords(enrichedAlerts, maxTelemetryBatch);
+          for (const batch of batches) {
+            await sender.sendOnDemand(TELEMETRY_CHANNEL_DETECTION_ALERTS, batch);
+          }
         }
-        await sender.sendOnDemand(TASK_METRICS_CHANNEL, [
```

Comment:
were the task metrics meant to be removed?

Reply:
No, my bad.